torch/testing/_internal/common_methods_invocations.py

```diff
@@ -635,6 +635,35 @@ def sample_inputs_gather(op_info, device, dtype, requires_grad):
                         0, torch.tensor(0, dtype=torch.int64, device=device))),
             )
 
+def sample_inputs_take_along_dim(op_info, device, dtype, requires_grad):
```

Comment:
How long do these new tests take in the CI?

Reply:
```
pytest test/test_ops.py -k _take_along_dim
========================================================================= test session starts ==========================================================================
platform linux -- Python 3.8.6, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /home/kshiteej/Pytorch/pytorch_take_along_axis
plugins: hypothesis-5.38.1
collected 9469 items / 9360 deselected / 109 selected                                                                                                                  

test/test_ops.py ............................ssss....ssss.....................................................................                                   [100%]
===================================================== 101 passed, 8 skipped, 9360 deselected, 4 warnings in 24.15s =====================================================
```

Reply:
@albanD we really need to reduce our grad{grad, }check time.

Reply:
For referece,

Durations with `pytest --durations`

cmd : `pytest test/test_ops.py -k _take_alo --durations=30`

Output
```
1.27s call     test/test_ops.py::TestOpInfoCUDA::test_supported_dtypes_take_along_dim_cuda_bfloat16
1.10s call     test/test_ops.py::TestGradientsCUDA::test_fn_gradgrad_take_along_dim_cuda_complex128
0.70s call     test/test_ops.py::TestCommonCUDA::test_variant_consistency_jit_take_along_dim_cuda_complex128
0.67s call     test/test_ops.py::TestCommonCUDA::test_variant_consistency_jit_take_along_dim_cuda_complex64
0.66s call     test/test_ops.py::TestCommonCUDA::test_variant_consistency_jit_take_along_dim_cuda_float32
0.64s call     test/test_ops.py::TestCommonCUDA::test_variant_consistency_jit_take_along_dim_cuda_float16
0.62s call     test/test_ops.py::TestCommonCUDA::test_variant_consistency_jit_take_along_dim_cuda_float64
0.54s call     test/test_ops.py::TestCommonCUDA::test_variant_consistency_jit_take_along_dim_cuda_bfloat16
0.49s call     test/test_ops.py::TestGradientsCUDA::test_fn_gradgrad_take_along_dim_cuda_float64
0.41s call     test/test_ops.py::TestGradientsCUDA::test_fn_grad_take_along_dim_cuda_complex128
0.34s call     test/test_ops.py::TestCommonCPU::test_variant_consistency_jit_take_along_dim_cpu_float64
0.34s call     test/test_ops.py::TestCommonCUDA::test_variant_consistency_jit_take_along_dim_cuda_uint8
0.34s call     test/test_ops.py::TestCommonCPU::test_variant_consistency_jit_take_along_dim_cpu_float32
0.33s call     test/test_ops.py::TestCommonCUDA::test_variant_consistency_jit_take_along_dim_cuda_int8
0.33s call     test/test_ops.py::TestCommonCUDA::test_variant_consistency_jit_take_along_dim_cuda_int16
0.32s call     test/test_ops.py::TestCommonCUDA::test_variant_consistency_jit_take_along_dim_cuda_int32
0.32s call     test/test_ops.py::TestCommonCPU::test_variant_consistency_jit_take_along_dim_cpu_complex64
0.32s call     test/test_ops.py::TestCommonCPU::test_variant_consistency_jit_take_along_dim_cpu_complex128
0.31s call     test/test_ops.py::TestCommonCUDA::test_variant_consistency_jit_take_along_dim_cuda_int64
0.30s call     test/test_ops.py::TestCommonCPU::test_variant_consistency_jit_take_along_dim_cpu_float16
0.30s call     test/test_ops.py::TestCommonCUDA::test_variant_consistency_jit_take_along_dim_cuda_bool
0.26s call     test/test_ops.py::TestOpInfoCUDA::test_supported_dtypes_take_along_dim_cuda_bool
0.26s call     test/test_ops.py::TestGradientsCPU::test_fn_gradgrad_take_along_dim_cpu_complex128
0.26s call     test/test_ops.py::TestCommonCUDA::test_out_take_along_dim_cuda_bfloat16
0.25s call     test/test_ops.py::TestCommonCUDA::test_out_take_along_dim_cuda_bool
0.24s call     test/test_ops.py::TestGradientsCUDA::test_fn_grad_take_along_dim_cuda_float64
0.23s call     test/test_ops.py::TestCommonCUDA::test_out_take_along_dim_cuda_complex128
0.23s call     test/test_ops.py::TestCommonCUDA::test_out_take_along_dim_cuda_float32
0.23s call     test/test_ops.py::TestCommonCUDA::test_out_take_along_dim_cuda_complex64
0.23s call     test/test_ops.py::TestCommonCUDA::test_out_take_along_dim_cuda_float16
===================================================== 101 passed, 8 skipped, 9781 deselected, 4 warnings in 24.91s =====================================================
```

`1.27s call     test/test_ops.py::TestOpInfoCUDA::test_supported_dtypes_take_along_dim_cuda_bfloat16` takes long probably because that is where CUDA is initialized first time.

Reply:
Thank you, @kshitij12345. I didn't realize the jit tests were often so slow. Maybe (not in this PR) we should consider pruning their dtypes even more.